tests/js/client/shell/api/cursor-noncluster.js

```diff
@@ -304,5 +306,239 @@ function testing_the_query_cacheSuite () {
+      }
+    },
+
+    test_testing_streaming_transaction_committed_set: function() {
+      let trx = null;
+      let res = arango.POST_RAW("/_api/cursor", {"query": `UPSERT { _key: "a" } INSERT { _key: "a", value: 0 } UPDATE { value: 0 } IN ${cn}`});
+      assertEqual(res.code, 201);
+      res = arango.POST_RAW("/_api/cursor", {"query": `FOR doc IN ${cn} FILTER doc._key == "a" RETURN doc`});
+      assertEqual(res.code, 201);
+      assertEqual(res.parsedBody.result[0].value, 0);
+      res = arango.POST_RAW("/_api/transaction/begin", {collections: {write: cn}});
+      assertEqual(res.code, 201);
+      assertNotUndefined(res.parsedBody.result.id);
+      trx = res.parsedBody.result.id;
+      assertNotNull(trx);
+      try {
+        res = arango.POST_RAW("/_api/cursor", {"query": `UPDATE { _key: "a", value: 1 } IN ${cn}`}, {"x-arango-trx-id": trx});
+
+        assertEqual(res.code, 201);
+        assertFalse(res.body.error);
+
+        res = arango.POST_RAW("/_api/cursor", {
+          "query": `FOR doc IN ${cn} FILTER doc._key == "a" RETURN doc`,
+          "cache": true
+        });
+        assertEqual(res.code, 201);
+        assertFalse(res.error);
+        assertEqual(res.parsedBody.result[0].value, 0);
+
+        res = arango.POST_RAW("/_api/cursor", {"query": `UPDATE { _key: "a", value: 2 } IN ${cn}`}, {"x-arango-trx-id": trx});
+        assertEqual(res.code, 201);
+        assertFalse(res.error);
+
+        res = arango.POST_RAW("/_api/cursor", {
+          "query": `FOR doc IN ${cn} FILTER doc._key == "a" RETURN doc`,
+          "cache": true
+        }, {"x-arango-trx-id": trx});
+        assertEqual(res.code, 201);
+        assertFalse(res.error);
+        assertEqual(res.parsedBody.result[0].value, 2);
+
+        res = arango.POST_RAW("/_api/cursor", {
+          "query": `FOR doc IN ${cn} FILTER doc._key == "a" RETURN doc`,
+          "cache": true
+        });
+        assertEqual(res.code, 201);
+        assertFalse(res.error);
+        assertEqual(res.parsedBody.result[0].value, 0);
+
+        res = arango.POST_RAW("/_api/cursor", {
+          "query": `FOR doc IN ${cn} FILTER doc._key == "a" RETURN doc`,
+          "cache": false
+        });
+        assertEqual(res.code, 201);
+        assertFalse(res.error);
+        assertEqual(res.parsedBody.result[0].value, 0);
+      } finally {
+        res = arango.PUT_RAW("/_api/transaction/" + encodeURIComponent(trx), {}, {});
+        assertEqual(res.code, 200);
+        assertFalse(res.error);
+        assertEqual(res.parsedBody.result.status, "committed");
+
+        res = arango.POST_RAW("/_api/cursor", {
+          "query": `FOR doc IN ${cn} FILTER doc._key == "a" RETURN doc`,
+          "cache": false
+        });
+        assertEqual(res.code, 201);
+        assertFalse(res.error);
+        assertEqual(res.parsedBody.result[0].value, 2);
+      }
+    },
+
+    test_testing_streaming_js_transaction_aborted_set: function() {
```

Comment:
The JavaScript Transactions are not Stream Transactions
```suggestion
    test_testing_js_transaction_aborted_set: function() {
```